src/Transformer/YamlTransformer.php

```diff
@@ -428,7 +428,7 @@ function ($document) use ($flag) {
         // and check their filename and maybe their document name, depending on the pattern.
         // We don't want to do any pattern matching after the first hash as the document name
         // is assumed to follow it.
-        $firstHash = strpos('#', $pattern);
+        $firstHash = strpos($pattern, '#');
         $documentName = false;
         if ($firstHash !== false) {
             $documentName = substr($pattern, $firstHash + 1);
```

Comment:
I suggest to update `$patternRegExp` so that instead of matching start of string (`^`) it allows matching any slash, so that `framework#name` matches `vendor/silverstripe/framework`.

Secondly, I also suggest we add a 'composername' key to $document, and match that as well as `filename`. We can get this from getting the composer.json content from one directory up, and getting the name key.

Reply:
hmm, so apparently configs in core have also been broken if they had used the `After: framework` rule.

For a quick-fix I'd prefer the first suggestion...
but long-term I would like to see the composer-name key implemented.

would @chillu like to give his input?

Reply:
Fix the bug, make `composername` an `impact/medium` improvement? I assume you mean `module/file#fragment` also supporting `vendor/module/file#fragment`? I think that would be consistent with the way it's described ("module" is the name of the module this YAML file is in."). See https://docs.silverstripe.org/en/4/developer_guides/configuration/configuration/

Reply:
ok, just make the regexp change I suggested (partial path matching) and we'll add composer name enhancement another time.

Reply:
I've raised https://github.com/silverstripe/silverstripe-config/issues/25

Reply:
@flamerohr you could even just use `\b` (word boundary) as a prefix for the pattern instead of `(^|/)` if you like.

Reply:
Cool, I guess I'd prefer `(^|[/\\\\])` for it's predictability :)
`\b` could mean a number of characters as well?

Reply:
looks good.